# Pertemuan 2 - Python, NumPy, dan Pandas

Nama = Abdul Zaki Al-Muttaqin
NIM : 240401010042
Kelas : IF403

## Tujuan Praktikum
Mempelajari penggunaan Python, NumPy, dan Pandas untuk membaca, mengolah, dan menganalisis data.

In [24]:
#Import Library
import pandas as pd
import numpy as np

In [25]:
#Load Dataset Titanic
import pandas as pd

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"

df = pd.read_csv(url)

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


# Q1
Menampilkan Jumlah baris, kolom, dan nama kolom dataset Titanic

In [26]:
#Kode Q1
print("Shape Dataset:")
print(df.shape)

print("\nDaftar Kolom:")
print(df.columns.tolist())

Shape Dataset:
(891, 12)

Daftar Kolom:
['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


Dataset Titanic memiliki 891 baris dan 12 kolom yang berisi informasi penumpang seperti nama, jenis kelamin, usia, kelas tiket, tarif, dan status keselamatan.

# Q2
Mencari missing values.

In [27]:
missing = df.isnull().sum()

persentase = (missing / len(df)) * 100

hasil_missing = pd.DataFrame({
    'Jumlah Missing': missing,
    'Persentase (%)': persentase
})

hasil_missing[hasil_missing['Jumlah Missing'] > 0]

,Jumlah Missing,Persentase (%)
Age,177,19.865320
Cabin,687,77.104377
Embarked,2,0.224467


Kolom Age, Cabin, dan Embarked memiliki data yang kosong (missing values). Kolom Cabin memiliki jumlah data kosong paling banyak sehingga perlu penanganan khusus sebelum digunakan untuk analisis lanjutan.

# Q3
Menghitung persentase penumpang yang selamat.

In [28]:
persen_selamat = df["Survived"].mean() * 100

print(f"Persentase penumpang yang selamat: {persen_selamat:.2f}%")

Persentase penumpang yang selamat: 38.38%


Sekitar 38% penumpang Titanic berhasil selamat, sedangkan sisanya tidak selamat

# Q4
Mencari jumlah penumpang wanitas pada kelas 1 dan rata rata usianya.

In [29]:
wanita_kelas1 = df[(df['Sex'] == 'female') & (df['Pclass'] == 1)]
jumlah = wanita_kelas1.shape[0]
rata_usia = wanita_kelas1['Age'].mean()

print("jumlah wanita kelas 1:", jumlah)
print("Rata-rata usia:", round(rata_usia, 2))

jumlah wanita kelas 1: 94
Rata-rata usia: 34.61


Penumpang wanita kelas 1 memiliki jumlah tertentu dan rata-rata usia yang menunjukkan kelompok penumpang kelas atas pada masa tersebut.

# Q5
Menghitung tingkat keselamatan berdasarkan kelas penumpang

In [30]:
hasil = pd.DataFrame({
    "Persentase Selamat (%)":
    df.groupby("Pclass")["Survived"].mean() * 100
})

hasil

,Persentase Selamat (%)
Pclass,
1,62.962963
2,47.282609
3,24.236253


Penumpang kelas 1 memiliki tingkat keselamatan tertinggi dibandingkan kelas lainnya. Hal ini menunjukkan bahwa status sosial dan lokasi kabin berpengaruh terhadap peluang keselamatan.

# Q6
Analisis tingkat keselamatan berdasarkan jenis kelamin

In [31]:
hasil_gender = pd.DataFrame({
    "Persentase Selamat (%)":
    df.groupby("Sex")["Survived"].mean() * 100 })

hasil_gender

,Persentase Selamat (%)
Sex,
female,74.203822
male,18.890815


Penumpang perempuan memiliki tingkat keselamatan yang jauh lebih tinggi dibandingkan laki-laki. Hal ini sesuai dengan kebijakan evakuasi Titanic yang mengutamakan perempuan dan anak-anak.

# Kesimpulan

Pada praktikum ini saya belajar menggunakan Python dan library Pandas untuk membaca dataset, menampilkan informasi data, serta melakukan analisis sederhana. Saya juga mengenal struktur DataFrame yang memudahkan proses pengolahan data.

Materi ini membantu saya memahami dasar-dasar pengolahan data yang nantinya akan digunakan pada proses analisis data dan machine learning.